### DISCUSSION TITLE
Properties

### DISCUSSION THREAD

**Discussion by ryanopily** [UNC-Project-Open-AAC/OS-DPI#116] [Ideas]
After some discussion, we think it may be important to be able to toggle a 'Display' component on and off with a grid button; more generally, dynamically change properties of a component based on some event. In this case, we'd want to toggle the scale of a Display component between the values 0 & 1 (really between 0.1 and 1, because scale is at least 0.1) whenever a designated toggle button is pressed on a grid.

I can't think of an elegant way of doing this at the moment - everything I think of seems to over complicate things. I just wanted to throw this idea out there.

,

**Comment by gbishop** [comment]
To make the Display disappear (what I think you want), you could have 2
tabs in a tab control, one with the display and one without. Then you
simply switch tabs to determine the visibility of the display.

This is in the spirit of "simulating" the desired behavior.

For the future, I would like to think about "computed properties" for
components. I played with it a little but wanted to push to get a complete
system first. Suppose that if a property value begins with an equal sign,
it is evaluated at run time. This is a bit like spreadsheet equations. Then
you could change anything with state variables. But this will require
careful thought and likely some internal restructuring.

That "programming portals" post I shared is also relevant here.
